Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-15E.lua

```diff
@@ -1171,4 +1171,590 @@ F_15E:defineMultipositionSwitch("R_LCP_LASER_DIG_4", 44, 3073, 73, 8, 0.1, "Rear
+F_15E:defineString("F_MPDR_PBLABEL_03", function() 
+    return F_MPDR_BUTTONS[3] 
+end, 10, "Front Right MPD Lables", "PB 03") 
+F_15E:defineString("F_MPDR_PBLABEL_04", function() 
+    return F_MPDR_BUTTONS[4] 
+end, 10, "Front Right MPD Lables", "PB 04") 
+F_15E:defineString("F_MPDR_PBLABEL_05", function() 
+    return F_MPDR_BUTTONS[5] 
+end, 10, "Front Right MPD Lables", "PB 05") 
+F_15E:defineString("F_MPDR_PBLABEL_06", function() 
+    return F_MPDR_BUTTONS[6] 
+end, 10, "Front Right MPD Lables", "PB 06") 
+F_15E:defineString("F_MPDR_PBLABEL_07", function() 
+    return F_MPDR_BUTTONS[7] 
+end, 10, "Front Right MPD Lables", "PB 07") 
+F_15E:defineString("F_MPDR_PBLABEL_08", function() 
+    return F_MPDR_BUTTONS[8] 
+end, 10, "Front Right MPD Lables", "PB 08") 
+F_15E:defineString("F_MPDR_PBLABEL_09", function() 
+    return F_MPDR_BUTTONS[9] 
+end, 10, "Front Right MPD Lables", "PB 09") 
+F_15E:defineString("F_MPDR_PBLABEL_10", function() 
+    return F_MPDR_BUTTONS[10] 
+end, 10, "Front Right MPD Lables", "PB 10") 
+F_15E:defineString("F_MPDR_PBLABEL_11", function() 
+    return F_MPDR_BUTTONS[11] 
+end, 10, "Front Right MPD Lables", "PB 11") 
+F_15E:defineString("F_MPDR_PBLABEL_12", function() 
+    return F_MPDR_BUTTONS[12] 
+end, 10, "Front Right MPD Lables", "PB 12") 
+F_15E:defineString("F_MPDR_PBLABEL_13", function() 
+    return F_MPDR_BUTTONS[13] 
+end, 10, "Front Right MPD Lables", "PB 13") 
+F_15E:defineString("F_MPDR_PBLABEL_14", function() 
+    return F_MPDR_BUTTONS[14] 
+end, 10, "Front Right MPD Lables", "PB 14") 
+F_15E:defineString("F_MPDR_PBLABEL_15", function() 
+    return F_MPDR_BUTTONS[15] 
+end, 10, "Front Right MPD Lables", "PB 15") 
+F_15E:defineString("F_MPDR_PBLABEL_16", function() 
+    return F_MPDR_BUTTONS[16] 
+end, 10, "Front Right MPD Lables", "PB 16") 
+F_15E:defineString("F_MPDR_PBLABEL_17", function() 
+    return F_MPDR_BUTTONS[17] 
+end, 10, "Front Right MPD Lables", "PB 17") 
+F_15E:defineString("F_MPDR_PBLABEL_18", function() 
+    return F_MPDR_BUTTONS[18] 
+end, 10, "Front Right MPD Lables", "PB 18") 
+F_15E:defineString("F_MPDR_PBLABEL_19", function() 
+    return F_MPDR_BUTTONS[19] 
+end, 10, "Front Right MPD Lables", "PB 19") 
+F_15E:defineString("F_MPDR_PBLABEL_20", function() 
+    return F_MPDR_BUTTONS[20] 
+end, 10, "Front Right MPD Lables", "PB 20")
+
+
+local R_MPCDL_BUTTONS = {}
+
+local R_MPDL_BUTTONS = {}
+
+local R_MPDR_BUTTONS = {}
+
+local R_MPCDR_BUTTONS = {}
+
+
+F_15E:addExportHook(function()
+	local input = list_indication(10)
+	if not input then
+		return
+	end
+	R_MPCDL_BUTTONS = parseMFDButtons(input)
+
+	input = list_indication(12)
+	if not input then
+		return
+	end
+	
+	R_MPDL_BUTTONS = parseMFDButtons(input)
+
+	input = list_indication(14)
+	if not input then
+		return
+	end
+	
+	R_MPDR_BUTTONS = parseMFDButtons(input)
+
+	input = list_indication(16)
+	if not input then
+		return
+	end
+	
+	R_MPCDR_BUTTONS = parseMFDButtons(input)
```

Comment:
This should probably get broken out into separate export hooks. This also has the potential for a bug - if e.g. indication 12 is nil, then it won't even try to process 14 or 16. I'd suggest a separate export hook for each one, especially since there is no shared data to benefit from.

The same goes for the front buttons.